pkg/shp/cmd/buildrun/create.go

```diff
@@ -74,14 +77,58 @@ func (c *CreateCommand) Run(params *params.Params, ioStreams *genericclioptions.
 	br := &buildv1alpha1.BuildRun{Spec: *c.buildRunSpec}
 	flags.SanitizeBuildRunSpec(&br.Spec)
 
+	if c.name == "" {
+		br.GenerateName = c.buildRunSpec.BuildRef.Name
+	}
+
 	clientset, err := params.ShipwrightClientSet()
 	if err != nil {
 		return err
 	}
-	if _, err = clientset.ShipwrightV1alpha1().BuildRuns(params.Namespace()).Create(c.cmd.Context(), br, metav1.CreateOptions{}); err != nil {
-		return err
+
+	build, err := clientset.ShipwrightV1alpha1().Builds(params.Namespace()).Get(c.cmd.Context(), c.buildRunSpec.BuildRef.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get referenced build %q: %w", c.buildRunSpec.BuildRef.Name, err)
 	}
-	fmt.Fprintf(ioStreams.Out, "BuildRun created %q for Build %q\n", c.name, br.Spec.BuildRef.Name)
+
+	// Local source code mode:
+	// Make sure to bundle the configured local source directory into an image
+	// bundle and push it to the provided target registry.
+	var digest name.Digest
+	if build.Spec.Source.BundleContainer.Image != "" {
+		if digest, err = bundle.Push(c.cmd.Context(), ioStreams, *c.localSourceDir, build.Spec.Source.BundleContainer.Image); err != nil {
+			return err
+		}
+	}
+
+	br, err = clientset.ShipwrightV1alpha1().BuildRuns(params.Namespace()).Create(c.cmd.Context(), br, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create BuildRun: %v", err)
+	}
+
+	fmt.Fprintf(ioStreams.Out, "BuildRun created %q for Build %q\n", br.Name, br.Spec.BuildRef.Name)
+
+	if *c.wait {
+		condition := func() (bool, error) {
+			br, err = clientset.ShipwrightV1alpha1().BuildRuns(params.Namespace()).Get(c.cmd.Context(), br.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			return br.Status.CompletionTime != nil, nil
+		}
+
+		if err := wait.PollUntil(5*time.Second, condition, c.Cmd().Context().Done()); err != nil {
```

Comment:
yeah in other parts of the CLI we are employing watches vs. polls for what IMO is analogous function

first blush, I'm inclined this follows that established pattern

@otaviof  do you agree?

/assign @otaviof

Reply:
I think we should follow the established `watch` pattern, as the following examples:
- https://github.com/shipwright-io/cli/blob/c8174b402cc94a1031d24622723d971180803475/pkg/shp/cmd/build/run.go#L94-L134
- https://github.com/shipwright-io/cli/blob/c8174b402cc94a1031d24622723d971180803475/pkg/shp/reactor/pod_watcher_test.go

Let me know if you need any help adopting this pattern, please.

Reply:
@gabemontero @otaviof I am sorry, it took me a bit to tackle this one. To my surprise, I realized that we have two independent code flows for creating a BuildRun:
- `shp build run my-build` or `shp build run --buildref-name my-build`
- `shp buildrun create --buildref-name my-build`

So I was looking into unifying this, because we should not end up at a state where we have to maintain two different code flows that practically do the same, or even worst, would end up with different flags and features. At first, I wanted to change the `build run` command to use the code from `buildrun create`, and vice verse, but I ended up with really a lot of code, which was hard to read and probably difficult to understand. So I decided to "externalize" the BuildRun creation into a separate function and both `build run` and `buildrun create` will share the same setup code (to setup the flags) and eventually call the common BuildRun creation function. With this, we can declutter the commands and keep them flag/settings centric while the actual implementation is in another function.

Reply:
ok @HeavyWombat just starting to catch up after the US holiday yesterday.

I see you pushed a new commit at the same time as this comment.  I'll start looking at it shortly.  But at a high level, I would agree that any consolidation of the 2 paths which wasn't there previously would be good.